File: classes/parte-7-paradigmas/115-funcional-ii-composicion-currying-y-aplicacion-parcial/implementaciones/php/main.php
<?php
$doblar = fn($x) => $x * 2;
$incrementar = fn($x) => $x + 1;
$compuesta = fn($x) => $incrementar($doblar($x));

$n = (int) trim(fgets(STDIN));
echo "resultado=" . $compuesta($n) . "\n";

File: classes/parte-7-paradigmas/115-funcional-ii-composicion-currying-y-aplicacion-parcial/implementaciones/rust/main.rs
use std::io::Read;

fn doblar(x: i64) -> i64 {
    x * 2
}

fn incrementar(x: i64) -> i64 {
    x + 1
}

fn main() {
    let mut s = String::new();
    std::io::stdin().read_to_string(&mut s).unwrap();
    let n: i64 = s.trim().parse().unwrap();
    println!("resultado={}", incrementar(doblar(n)));
}
